exporter/collector/logs.go

```diff
@@ -109,8 +116,51 @@ func (l *LogsExporter) Shutdown(ctx context.Context) error {
 }
 
 func (l *LogsExporter) PushLogs(ctx context.Context, ld plog.Logs) error {
+	entries, err := l.mapper.createEntries(ld)
+	if err != nil {
+		return err
+	}
+
+	errors := []error{}
+	entry := 0
+	currentBatchSize := 0
+	// Send entries in WriteRequest chunks
+	for len(entries) > 0 {
+		// TODO(damemi): Add integration test for batch request processing
+		if entry < len(entries) && currentBatchSize+proto.Size(entries[entry]) < defaultMaxRequestSize {
+			// if adding the current entry to the current batch doesn't go over the request size,
+			// increase the index and account for the new request size, then continue
+			currentBatchSize = currentBatchSize + proto.Size(entries[entry])
+			entry++
+			continue
+		}
+
+		// if the current entry goes over the request size (or we have gone over every entry, i.e. index=len),
+		// write the list up to but not including the current entry's index
+		_, err := l.writeLogEntries(ctx, entries[:entry])
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		// if there are still entries to parse, reassign entries to a
+		// shortened list starting at the current index
+		if entry < len(entries) {
```

Comment:
Is this check is necessary?  If there aren't any entries left to parse, won't `len(entries) > 0` above return false?

Reply:
Not necessary, think I was off by one protecting for panics